vanguard_driver.py

```diff
@@ -378,12 +389,29 @@ def run(args: VanguardDriverOpts, lang: Platform):
     )
 
     if results:
-        print("----VANGUARD REPORT----", file=sys.stdout)
-        print(results, file=sys.stdout)
+        args.logger.info("----VANGUARD REPORT----")
+        for line in results.splitlines():
+            args.logger.info(line)
+
+
+def _configure_logging(logger: logging.Logger, log_level: int):
+    logger.setLevel(log_level)
+    # WARNING severity and above is logged to stderr.
+    err_handler = logging.StreamHandler(stream=sys.stderr)
+    err_handler.setLevel(logging.WARNING)
+    err_handler.setFormatter(
+        logging.Formatter(fmt="{levelname}: {message}", style="{"))
```

Comment:
```suggestion
        logging.Formatter(fmt="{asctime} [{levelname}]: {message}", style="{"))
```

IMO always good to have the date in a log

Reply:
I generally agree with you, but the current mode is meant for interactive command line use; timestamps will clutter up the log. There's nothing preventing the user from attaching timestamps by [piping to external tools](https://unix.stackexchange.com/questions/26728/prepending-a-timestamp-to-each-line-of-output-from-a-command), so this is a matter of whether we want to show the timestamp by default.

I'll soon be submitting a follow up PR that has the json logging mode. As the json mode is not meant for human consumption, it includes more details like timestamps and source location by default.

Reply:
I see, I was not aware of that. In that case makes sense to do it that way